### PR TITLE
[DETR] Improve code examples

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

As a follow-up of #17786, this PR improves the code examples of DETR to showcase the `post_process` methods.

### CODE REVIEW

File: src/transformers/models/detr/modeling_detr.py [hunk 1]
```diff
@@ -1237,8 +1237,15 @@ def forward(
...
         >>> feature_extractor = DetrFeatureExtractor.from_pretrained("facebook/detr-resnet-50")
         >>> model = DetrModel.from_pretrained("facebook/detr-resnet-50")
+
+        >>> # prepare image for the model
         >>> inputs = feature_extractor(images=image, return_tensors="pt")
+
+        >>> # forward pass
         >>> outputs = model(**inputs)
+
+        >>> # the last hidden states are the final query embeddings of the Transformer decoder
```
Comment (alaradirik) [inline src/transformers/models/detr/modeling_detr.py:1247]: Looks great! Comment lines make it easier to follow examples

File: src/transformers/models/detr/modeling_detr.py [hunk 2]
```diff
@@ -1553,29 +1567,36 @@ def forward(
 
         ```python
         >>> from transformers import DetrFeatureExtractor, DetrForSegmentation
+        >>> from transformers.models.detr.feature_extraction_detr import rgb_to_id
+        >>> import torch
         >>> from PIL import Image
+        >>> import numpy
         >>> import requests
+        >>> import io
```
Comment (alaradirik) [inline src/transformers/models/detr/modeling_detr.py:1575]: This is not very important but importing low-level libraries first improves readability
```
import io
import requests
from PIL import Image
import torch
import numpy

from transformers import DetrFeatureExtractor, DetrForSegmentation
from transformers.models.detr.feature_extraction_detr import rgb_to_id
```

File: src/transformers/models/detr/modeling_detr.py [hunk 3]
```diff
@@ -1553,29 +1567,36 @@ def forward(
...
-
-        >>> masks = outputs.pred_masks
-        >>> list(masks.shape)
-        [1, 100, 200, 267]
+
+        >>> # use the `post_process_panoptic` method of `DetrFeatureExtractor` to convert to COCO format
+        >>> processed_sizes = torch.as_tensor(inputs["pixel_values"].shape[-2:]).unsqueeze(0)
+        >>> result = feature_extractor.post_process_panoptic(outputs, processed_sizes)[0]
+
+        >>> # the segmentation is stored in a special-format png
```
Comment (alaradirik) [inline src/transformers/models/detr/modeling_detr.py:1593]: Great, makes it much easier to understand the segmentation output

**Review by alaradirik — COMMENTED** [review]
Looks great! The example snippets were my main source of information before joining the team, they really make a difference

**Review by alaradirik — APPROVED** [review]

**Review by alaradirik — APPROVED** [review]

File: src/transformers/models/detr/modeling_detr.py [hunk 4]
```diff
@@ -1400,17 +1408,23 @@ def forward(
...
+        >>> # convert outputs (bounding boxes and class logits) to COCO API
+        >>> target_sizes = torch.tensor([image.size[::-1]])
+        >>> results = feature_extractor.post_process(outputs, target_sizes=target_sizes)[0]
+
+        >>> for score, label, box in zip(results["scores"], results["labels"], results["boxes"]):
+        ...     box = [round(i, 2) for i in box.tolist()]
+        ...     # let's only keep detections with score > 0.9
+        ...     if score > 0.9:
+        ...         print(
+        ...             f"Detected {model.config.id2label[label.item()]} with confidence {round(score.item(), 3)} at location {box}"
```
Comment (sgugger) [inline src/transformers/models/detr/modeling_detr.py:1421]:
```suggestion
        ...             f"Detected {model.config.id2label[label.item()]} with confidence "
        ...             f"{round(score.item(), 3)} at location {box}."
```
Will be easier to read if the line is not too long.

**Review by sgugger — APPROVED** [review]
